by-language/python/blob.py

```diff
@@ -0,0 +1,255 @@
+    418a0143404fb2da8a1464ab721f6d5fb50c3b96
+
+    # Download an item from the BLOB store.
+    python blob.py download 418a0143404fb2da8a1464ab721f6d5fb50c3b96
+
+Full command line example, without defining environment variables::
+
+    python blob.py \
+        --url=http://crate@localhost:4200/ --container=testdrive \
+        upload /path/to/file
+
+## References
+
+- https://cratedb.com/docs/python/en/latest/blobs.html
+- https://cratedb.com/docs/crate/reference/en/latest/general/blobs.html
+- https://cratedb.com/docs/crate/reference/en/latest/sql/statements/create-blob-table.html
+"""
+import io
+import logging
+import os
+import sys
+import typing as t
+from argparse import ArgumentError, ArgumentParser
+from pathlib import Path
+
+from crate import client
+from crate.client.blob import BlobContainer
+from crate.client.connection import Connection
+from crate.client.exceptions import ProgrammingError
+
+logger = logging.getLogger(__name__)
+
+
+class CrateDbBlobContainer:
+    def __init__(self, url: str, name: str):
+        self.url = url
+        self.name = name
+        self.connection: Connection = None
+        self.container: BlobContainer = None
+
+    def connect(self):
+        self.connection = client.connect(self.url)
+        self.provision()
+        self.container = self.connection.get_blob_container(self.name)
+
+    def disconnect(self):
+        self.connection.close()
+
+    def upload(self, payload: t.Union[bytes, bytearray]) -> str:
+        """
+        Upload an item to the BLOB store.
+
+        - https://cratedb.com/docs/python/en/latest/blobs.html#upload-blobs
+        - https://cratedb.com/docs/crate/reference/en/5.4/general/blobs.html#uploading
+        """
+        file = io.BytesIO(payload)
+        return self.container.put(file)
+
+    def download(self, digest: str) -> bytes:
+        """
+        Download an item from the BLOB store.
+
+        - https://cratedb.com/docs/python/en/latest/blobs.html#retrieve-blobs
+        - https://cratedb.com/docs/crate/reference/en/5.4/general/blobs.html#downloading
+        """
+
+        payload = b""
+
+        for chunk in self.container.get(digest):
+            payload += chunk
+
+        return payload
+
+    def delete(self, digest: str) -> bool:
+        """
+        Delete an item from the BLOB store.
+        """
+        return self.container.delete(digest)
+
+    def provision(self):
+        """
+        Create a new table for storing Binary Large Objects (BLOBs).
+
+        - https://cratedb.com/docs/crate/reference/en/latest/sql/statements/create-blob-table.html
+        """
+        try:
+            self.run_sql(f'CREATE BLOB TABLE "{self.name}";')
+        except ProgrammingError as ex:
+            if "RelationAlreadyExists" not in ex.message:
+                raise
+
+    def refresh(self):
+        """
+        Optionally synchronize data after write operations.
+        """
+        self.run_sql(f'REFRESH TABLE "{self.name}";')
```

Comment:
looks unused, remove it?
Otherwise I think it would have to use the `blob` schema, such as `REFRESH TABLE blob."{self.name}";`

Reply:
Yeah, I also recognized this detail yesterday. Thank you. So, effectively, there is no way to run a BLOB table in a different schema? It is _always_ `blob`?

Reply:
Yes it's always `blob`, see also https://cratedb.com/docs/crate/reference/en/5.4/sql/statements/create-blob-table.html (syntax does not support to define a schema name).